src/realm/CMakeLists.txt

```diff
@@ -58,219 +63,32 @@ set(REALM_SOURCES
     utilities.cpp
     version.cpp
     views.cpp
-) # REALM_SOURCES
+) # add_library(realm)
+add_library(Realm::Core ALIAS Core)
```

Comment:
I believe the fact that this `add_library` invocation no longer includes any headers will result in the headers not being present in the generated Xcode project / Visual Studio solutions.

Reply:
I can't speak about Xcode, but in Visual Studio we don't care about including headers in targets for the sake of displaying them in the IDE because we develop with Visual Studio's Open Folder workflow, instead of generating a solution with CMake manually and opening that.

Reply:
Perhaps you can walk me through that at some point (or update the README to describe it?), since whenever I deal with Core on Windows I generate the solution via CMake and open it.

It is definitely an issue for Xcode, though.

Reply:
Visual Studio can now simply open CMake projects so simply open the right folder and it should do the rest.
As for Xcode, I thought VS was the only offender in the past, so once I checked with Lasse that VS support for CMake now works fine, I considered it done. Too bad really... I'll have to reintroduce the headers in there.

Reply:
Just to let you know, I've seen recent versions of VS fail to consider headers as a target dependency when using the Open Folder workflow with CMake, and they wouldn't be present in the Ctrl+T quick open dialog. Unless this has changed very recently (@emanuelez indicated yesterday that it has, I haven't verified yet), it makes sense to include the headers in the target. It shouldn't be a big deal either.